minijinja/src/compiler/parser.rs

```diff
@@ -1057,7 +1073,11 @@ impl<'a> Parser<'a> {
                         return Ok(rv);
                     }
                     rv.push(ok!(self.parse_stmt()));
-                    expect_token!(self, Token::BlockEnd(..), "end of block");
+
+                    // End of block may not exist, for example with do block.
+                    if let Some((Token::BlockEnd(..), _)) = ok!(self.stream.current()) {
```

Comment:
I'm not sure if this is the best way to check, or would lead to user confusion.

If they are using enddo, it will error anyway.